QuestionBot/QuestionBot/Model/MessageEmitter.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace QuestionBot.Model {
+    public class MessageEmitter : IMessageEmitter {
+        private string _lineInput = "";
+        private static IList<IMessageListener> _listeners = new List<IMessageListener>();
+
+        public void Start() {
+            while( _lineInput != "/exitQuestionbot" ) {
+                _lineInput = Console.ReadLine();
+                NotifyAllListeners( _lineInput );
+            }
+        }
+
+        public void Add( IMessageListener listener ) {
+            _listeners.Add( listener );
+        }
+
+        public static void NotifyAllListeners( string newInput ) {
+            for( int i = 0; i < _listeners.Count(); i++ ) {
```

Comment:
There's nothing wrong with this loop but it's more idiomatic to use a `foreach` loop. The reason for this is because we avoid the unnecessary `i` counter variable and `foreach` reads better. My basic rule is that unless I need `i` for something other than index, I'll use a `foreach`.